### ISSUE TITLE
Unable to send TRACE request with payload and with an outgoing proxy

### ISSUE THREAD

**Issue by benken-parasoft** [zaproxy/zaproxy#7578]
### Describe the bug

After upgrading from ZAP 2.11.1 to 2.12.0, I stopped receiving some alerts when running an active scan.  In particular, I no longer get alerts from the "Proxy Disclosure" scan rule which I normally see after configuring proxy settings (under Options > Network > Connection).  The log shows the following runtime exception being thrown from the network add-on from ProxyDisclosureScanRule:
```
java.lang.IllegalStateException: TRACE requests may not include an entity
        at org.apache.hc.core5.http.io.support.ClassicRequestBuilder.build(ClassicRequestBuilder.java:377) ~[?:?]
        at org.apache.hc.client5.http.impl.classic.ProtocolExec.execute(ProtocolExec.java:130) ~[?:?]
        at org.apache.hc.client5.http.impl.classic.ExecChainElement.execute(ExecChainElement.java:51) ~[?:?]
        at org.apache.hc.client5.http.impl.classic.ZapHttpRequestRetryExec.execute(ZapHttpRequestRetryExec.java:81) ~[?:?]
        at org.apache.hc.client5.http.impl.classic.ExecChainElement.execute(ExecChainElement.java:51) ~[?:?]
        at org.apache.hc.client5.http.impl.classic.ZapInternalHttpClient.doExecute(ZapInternalHttpClient.java:158) ~[?:?]
        at org.apache.hc.client5.http.impl.classic.CloseableHttpClient.execute(CloseableHttpClient.java:245) ~[?:?]
        at org.apache.hc.client5.http.impl.classic.CloseableHttpClient.execute(CloseableHttpClient.java:188) ~[?:?]
        at org.zaproxy.addon.network.internal.client.apachev5.HttpSenderApache.sendImpl0(HttpSenderApache.java:399) ~[?:?]
        at org.zaproxy.addon.network.internal.client.apachev5.HttpSenderApache.sendImpl(HttpSenderApache.java:293) ~[?:?]
        at org.zaproxy.addon.network.internal.client.apachev5.HttpSenderApache.sendImpl(HttpSenderApache.java:104) ~[?:?]
        at org.zaproxy.addon.network.internal.client.BaseHttpSender.sendAuthenticated(BaseHttpSender.java:298) ~[?:?]
        at org.zaproxy.addon.network.internal.client.BaseHttpSender.sendNoRedirections(BaseHttpSender.java:266) ~[?:?]
        at org.zaproxy.addon.network.internal.client.BaseHttpSender.send(BaseHttpSender.java:222) ~[?:?]
        at org.zaproxy.addon.network.internal.client.BaseHttpSender.sendAndReceive(BaseHttpSender.java:193) ~[?:?]
        at org.zaproxy.addon.network.internal.client.BaseHttpSender.sendAndReceive(BaseHttpSender.java:57) ~[?:?]
        at org.parosproxy.paros.network.HttpSender.sendAndReceive(HttpSender.java:316) ~[zap-2.12.0.jar:2.12.0]
        at org.parosproxy.paros.core.scanner.AbstractPlugin.sendAndReceive(AbstractPlugin.java:319) ~[zap-2.12.0.jar:2.12.0]
        at org.parosproxy.paros.core.scanner.AbstractPlugin.sendAndReceive(AbstractPlugin.java:250) ~[zap-2.12.0.jar:2.12.0]
        at org.zaproxy.zap.extension.ascanrulesBeta.ProxyDisclosureScanRule.scan(ProxyDisclosureScanRule.java:264) ~[?:?]
        at org.parosproxy.paros.core.scanner.AbstractPlugin.run(AbstractPlugin.java:361) ~[zap-2.12.0.jar:2.12.0]
        at java.lang.Thread.run(Thread.java:834) ~[?:?]
```

I also see a similar thing happening from InsecureHttpMethodScanRule:
```
java.lang.IllegalStateException: TRACE requests may not include an entity
        at org.apache.hc.core5.http.io.support.ClassicRequestBuilder.build(ClassicRequestBuilder.java:377) ~[?:?]
        at org.apache.hc.client5.http.impl.classic.ProtocolExec.execute(ProtocolExec.java:130) ~[?:?]
        at org.apache.hc.client5.http.impl.classic.ExecChainElement.execute(ExecChainElement.java:51) ~[?:?]
        at org.apache.hc.client5.http.impl.classic.ZapHttpRequestRetryExec.execute(ZapHttpRequestRetryExec.java:81) ~[?:?]
        at org.apache.hc.client5.http.impl.classic.ExecChainElement.execute(ExecChainElement.java:51) ~[?:?]
        at org.apache.hc.client5.http.impl.classic.ZapInternalHttpClient.doExecute(ZapInternalHttpClient.java:158) ~[?:?]
        at org.apache.hc.client5.http.impl.classic.CloseableHttpClient.execute(CloseableHttpClient.java:245) ~[?:?]
        at org.apache.hc.client5.http.impl.classic.CloseableHttpClient.execute(CloseableHttpClient.java:188) ~[?:?]
        at org.zaproxy.addon.network.internal.client.apachev5.HttpSenderApache.sendImpl0(HttpSenderApache.java:399) ~[?:?]
        at org.zaproxy.addon.network.internal.client.apachev5.HttpSenderApache.sendImpl(HttpSenderApache.java:293) ~[?:?]
        at org.zaproxy.addon.network.internal.client.apachev5.HttpSenderApache.sendImpl(HttpSenderApache.java:104) ~[?:?]
        at org.zaproxy.addon.network.internal.client.BaseHttpSender.sendAuthenticated(BaseHttpSender.java:298) ~[?:?]
        at org.zaproxy.addon.network.internal.client.BaseHttpSender.sendNoRedirections(BaseHttpSender.java:266) ~[?:?]
        at org.zaproxy.addon.network.internal.client.BaseHttpSender.send(BaseHttpSender.java:222) ~[?:?]
        at org.zaproxy.addon.network.internal.client.BaseHttpSender.sendAndReceive(BaseHttpSender.java:193) ~[?:?]
        at org.zaproxy.addon.network.internal.client.BaseHttpSender.sendAndReceive(BaseHttpSender.java:57) ~[?:?]
        at org.parosproxy.paros.network.HttpSender.sendAndReceive(HttpSender.java:316) ~[zap-2.12.0.jar:2.12.0]
        at org.parosproxy.paros.core.scanner.AbstractPlugin.sendAndReceive(AbstractPlugin.java:319) ~[zap-2.12.0.jar:2.12.0]
        at org.parosproxy.paros.core.scanner.AbstractPlugin.sendAndReceive(AbstractPlugin.java:250) ~[zap-2.12.0.jar:2.12.0]
        at org.zaproxy.zap.extension.ascanrulesBeta.InsecureHttpMethodScanRule.testTraceOrTrack(InsecureHttpMethodScanRule.java:375) ~[?:?]
        at org.zaproxy.zap.extension.ascanrulesBeta.InsecureHttpMethodScanRule.scan(InsecureHttpMethodScanRule.java:283) ~[?:?]
        at org.parosproxy.paros.core.scanner.AbstractPlugin.run(AbstractPlugin.java:361) ~[zap-2.12.0.jar:2.12.0]
        at java.lang.Thread.run(Thread.java:834) ~[?:?]
```

### Steps to reproduce the behavior

1. Install ZAP "Core Cross Platform Package".
2. Start zap.sh then install the alpha and beta active scan rules and restart.
3. Configure proxy settings under "Options > Network > Connection".  I had tried a couple different proxy servers including an old Squid proxy server.
4. Run an active scan against some URL.

### Expected behavior

In ZAP 2.11.1 I would get a "Proxy Disclosure" alert.  In ZAP 2.12.0 I now get nothing.

### Software versions

OWASP ZAP
Version: 2.12.0

Installed Add-ons: [[id=ascanrules, version=49.0.0],
[id=ascanrulesAlpha, version=41.0.0], [id=ascanrulesBeta,
version=43.0.0], [id=bruteforce, version=12.0.0],
[id=callhome, version=0.5.0], [id=commonlib,
version=1.11.0], [id=database, version=0.1.0], [id=diff,
version=12.0.0], [id=gettingStarted, version=14.0.0],
[id=help, version=15.0.0], [id=invoke, version=12.0.0],
[id=network, version=0.4.0], [id=oast, version=0.13.0],
[id=onlineMenu, version=10.0.0], [id=parasoft,
version=1.0.0], [id=pscanrules, version=44.0.0],
[id=pscanrulesAlpha, version=37.0.0], [id=pscanrulesBeta,
version=31.0.0], [id=quickstart, version=35.0.0],
[id=reports, version=0.16.0], [id=reveal, version=5.0.0],
[id=tips, version=10.0.0]]

Operating System: Linux
Architecture: amd64
Java Version: Oracle Corporation 11.0.16.1
System's Locale: en
Display Locale: en
Format Locale: en
Default Charset: UTF-8

### Errors from the zap.log file

From ProxyDisclosureScanRule:
```
java.lang.IllegalStateException: TRACE requests may not include an entity
        at org.apache.hc.core5.http.io.support.ClassicRequestBuilder.build(ClassicRequestBuilder.java:377) ~[?:?]
        at org.apache.hc.client5.http.impl.classic.ProtocolExec.execute(ProtocolExec.java:130) ~[?:?]
        at org.apache.hc.client5.http.impl.classic.ExecChainElement.execute(ExecChainElement.java:51) ~[?:?]
        at org.apache.hc.client5.http.impl.classic.ZapHttpRequestRetryExec.execute(ZapHttpRequestRetryExec.java:81) ~[?:?]
        at org.apache.hc.client5.http.impl.classic.ExecChainElement.execute(ExecChainElement.java:51) ~[?:?]
        at org.apache.hc.client5.http.impl.classic.ZapInternalHttpClient.doExecute(ZapInternalHttpClient.java:158) ~[?:?]
        at org.apache.hc.client5.http.impl.classic.CloseableHttpClient.execute(CloseableHttpClient.java:245) ~[?:?]
        at org.apache.hc.client5.http.impl.classic.CloseableHttpClient.execute(CloseableHttpClient.java:188) ~[?:?]
        at org.zaproxy.addon.network.internal.client.apachev5.HttpSenderApache.sendImpl0(HttpSenderApache.java:399) ~[?:?]
        at org.zaproxy.addon.network.internal.client.apachev5.HttpSenderApache.sendImpl(HttpSenderApache.java:293) ~[?:?]
        at org.zaproxy.addon.network.internal.client.apachev5.HttpSenderApache.sendImpl(HttpSenderApache.java:104) ~[?:?]
        at org.zaproxy.addon.network.internal.client.BaseHttpSender.sendAuthenticated(BaseHttpSender.java:298) ~[?:?]
        at org.zaproxy.addon.network.internal.client.BaseHttpSender.sendNoRedirections(BaseHttpSender.java:266) ~[?:?]
        at org.zaproxy.addon.network.internal.client.BaseHttpSender.send(BaseHttpSender.java:222) ~[?:?]
        at org.zaproxy.addon.network.internal.client.BaseHttpSender.sendAndReceive(BaseHttpSender.java:193) ~[?:?]
        at org.zaproxy.addon.network.internal.client.BaseHttpSender.sendAndReceive(BaseHttpSender.java:57) ~[?:?]
        at org.parosproxy.paros.network.HttpSender.sendAndReceive(HttpSender.java:316) ~[zap-2.12.0.jar:2.12.0]
        at org.parosproxy.paros.core.scanner.AbstractPlugin.sendAndReceive(AbstractPlugin.java:319) ~[zap-2.12.0.jar:2.12.0]
        at org.parosproxy.paros.core.scanner.AbstractPlugin.sendAndReceive(AbstractPlugin.java:250) ~[zap-2.12.0.jar:2.12.0]
        at org.zaproxy.zap.extension.ascanrulesBeta.ProxyDisclosureScanRule.scan(ProxyDisclosureScanRule.java:264) ~[?:?]
        at org.parosproxy.paros.core.scanner.AbstractPlugin.run(AbstractPlugin.java:361) ~[zap-2.12.0.jar:2.12.0]
        at java.lang.Thread.run(Thread.java:834) ~[?:?]
```

From InsecureHttpMethodScanRule:
```
java.lang.IllegalStateException: TRACE requests may not include an entity
        at org.apache.hc.core5.http.io.support.ClassicRequestBuilder.build(ClassicRequestBuilder.java:377) ~[?:?]
        at org.apache.hc.client5.http.impl.classic.ProtocolExec.execute(ProtocolExec.java:130) ~[?:?]
        at org.apache.hc.client5.http.impl.classic.ExecChainElement.execute(ExecChainElement.java:51) ~[?:?]
        at org.apache.hc.client5.http.impl.classic.ZapHttpRequestRetryExec.execute(ZapHttpRequestRetryExec.java:81) ~[?:?]
        at org.apache.hc.client5.http.impl.classic.ExecChainElement.execute(ExecChainElement.java:51) ~[?:?]
        at org.apache.hc.client5.http.impl.classic.ZapInternalHttpClient.doExecute(ZapInternalHttpClient.java:158) ~[?:?]
        at org.apache.hc.client5.http.impl.classic.CloseableHttpClient.execute(CloseableHttpClient.java:245) ~[?:?]
        at org.apache.hc.client5.http.impl.classic.CloseableHttpClient.execute(CloseableHttpClient.java:188) ~[?:?]
        at org.zaproxy.addon.network.internal.client.apachev5.HttpSenderApache.sendImpl0(HttpSenderApache.java:399) ~[?:?]
        at org.zaproxy.addon.network.internal.client.apachev5.HttpSenderApache.sendImpl(HttpSenderApache.java:293) ~[?:?]
        at org.zaproxy.addon.network.internal.client.apachev5.HttpSenderApache.sendImpl(HttpSenderApache.java:104) ~[?:?]
        at org.zaproxy.addon.network.internal.client.BaseHttpSender.sendAuthenticated(BaseHttpSender.java:298) ~[?:?]
        at org.zaproxy.addon.network.internal.client.BaseHttpSender.sendNoRedirections(BaseHttpSender.java:266) ~[?:?]
        at org.zaproxy.addon.network.internal.client.BaseHttpSender.send(BaseHttpSender.java:222) ~[?:?]
        at org.zaproxy.addon.network.internal.client.BaseHttpSender.sendAndReceive(BaseHttpSender.java:193) ~[?:?]
        at org.zaproxy.addon.network.internal.client.BaseHttpSender.sendAndReceive(BaseHttpSender.java:57) ~[?:?]
        at org.parosproxy.paros.network.HttpSender.sendAndReceive(HttpSender.java:316) ~[zap-2.12.0.jar:2.12.0]
        at org.parosproxy.paros.core.scanner.AbstractPlugin.sendAndReceive(AbstractPlugin.java:319) ~[zap-2.12.0.jar:2.12.0]
        at org.parosproxy.paros.core.scanner.AbstractPlugin.sendAndReceive(AbstractPlugin.java:250) ~[zap-2.12.0.jar:2.12.0]
        at org.zaproxy.zap.extension.ascanrulesBeta.InsecureHttpMethodScanRule.testTraceOrTrack(InsecureHttpMethodScanRule.java:375) ~[?:?]
        at org.zaproxy.zap.extension.ascanrulesBeta.InsecureHttpMethodScanRule.scan(InsecureHttpMethodScanRule.java:283) ~[?:?]
        at org.parosproxy.paros.core.scanner.AbstractPlugin.run(AbstractPlugin.java:361) ~[zap-2.12.0.jar:2.12.0]
        at java.lang.Thread.run(Thread.java:834) ~[?:?]
```

### Additional context

I am also running with "network-beta-0.4.0.zap" built from master to work around #7566.

### Would you like to help fix this issue?

- [ ] Yes

**Comment by thc202** [comment]
Could you check again with latest main?

**Comment by benken-parasoft** [comment]
I noticed that version 0.4.0 of the network add-on was published to the marketplace.  I no longer see the mentioned IllegalStateException showing up in the logs.

However, I am still not seeing any "Proxy Disclosure" alert being reported.  I took a look at the attack messages generated by this rule in ZAP 2.11.1 and in 2.12.0.  The requests are basically the same except that in 2.12.0 the "Proxy-Authorization" header is missing.  I see the "Proxy-Authorization" header in requests from other active scan rules but not the "Proxy Disclosure" rule.  This looks like another regression.  Should this issue be re-opened or should I file a new issue?
